### PR TITLE
feat : Upgrade to latest sepc256k1 (version 0.30.0)

### LINKED ISSUE #3482: Upgrade to latest `sepc256k1` (version `0.30.0`)

### PR DESCRIPTION
**Pull request by mahmudsudo**
closes #3482

### CODE REVIEW

File: bitcoin/examples/ecdsa-psbt.rs [hunk 1]
```diff
@@ -284,9 +286,13 @@ fn previous_output() -> TxOut {
 struct Error(Box<dyn std::error::Error>);
 
 impl<T: std::error::Error + 'static> From<T> for Error {
-    fn from(e: T) -> Self { Error(Box::new(e)) }
+    fn from(e: T) -> Self {
+        Error(Box::new(e))
+    }
```
Comment (tcharding) [inline bitcoin/examples/ecdsa-psbt.rs:291]: Please don't do formatting changes. If you didn't mean this to happen you might have your editor configured to format on save and you might be using the stable toolchain. We use a nightly toolchain to format.

Comment (mahmudsudo) [reply bitcoin/examples/ecdsa-psbt.rs]: i have reversed this with the nightly toolchain

**Comment by tcharding** [comment]
Can you squash all the commits please. Also you'll need to run `just update-lock-files` to update the two lock files - that should clear your CI failures.

**Comment by tcharding** [comment]
And since you are new, I might as well mention; we don't typically use `feat: ...` style, but you can if you really want to. Thanks for your contribution!

**Comment by mahmudsudo** [comment]
should i close this pr and reopen another ?

File: bitcoin/src/blockdata/script/witness_program.rs [hunk 2]
```diff
@@ -172,13 +174,15 @@ mod tests {
...
+        assert!(WitnessProgram::new(WitnessVersion::V15, &arbitrary_bytes).is_err());
+        // Arbitrary version
     }
 
     #[test]
     fn witness_program_is_too_long() {
         let arbitrary_bytes = [0x00; MAX_SIZE + 1];
-        assert!(WitnessProgram::new(WitnessVersion::V15, &arbitrary_bytes).is_err()); // Arbitrary version
+        assert!(WitnessProgram::new(WitnessVersion::V15, &arbitrary_bytes).is_err());
+        // Arbitrary version
```
Comment (yancyribbens) [inline bitcoin/src/blockdata/script/witness_program.rs:185]: nit: this comment doesn't make much sense where it's been moved.

**Comment by yancyribbens** [comment]
> should i close this pr and reopen another ?

I think it would make more sense to update this PR with @tcharding suggestions instead of opening a new PR.

**Comment by yancyribbens** [comment]
Thanks for fixing the comment, although it looks like the lock files still need to be updated.

**Comment by yancyribbens** [comment]
~although it looks like the lock files still need to be updated.~

Looking closer at the CI failure, it doesn't look like a problem with the lock file but rather after this change, the test `private_key_debug_is_obfuscated` is now failing which seems legitimate.

**Comment by mahmudsudo** [comment]
> ~although it looks like the lock files still need to be updated.~
> 
> Looking closer at the CI failure, it doesn't look like a problem with the lock file but rather after this change, the test `private_key_debug_is_obfuscated` is now failing which seems legitimate.

it seems the test values are different hence the reason they are failing , am i allowed to edit the values to allow them pass

**Comment by tcharding** [comment]
Yes you can change it to 

```rust
        let want = "PrivateKey { compressed: true, network: Test, inner: SecretKey(#7217ac58fbad8880) }";
```

There are currently two tests with the same name, you can remove them both and use this
```rust
    #[test]
    fn private_key_debug_is_obfuscated() {
        let sk =
            "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy".parse::<PrivateKey>().unwrap();
        let want =
            "PrivateKey { compressed: true, network: Test, inner: SecretKey(#7217ac58fbad8880) }";
        let got = format!("{:?}", sk);
        assert_eq!(got, want)
    }
```

The reason is we patched `impl_display_secret` macro during last scep release. Can you add a comment saying as such to the commit log please.

**Comment by mahmudsudo** [comment]
> Yes you can change it to
> 
> ```rust
>         let want = "PrivateKey { compressed: true, network: Test, inner: SecretKey(#7217ac58fbad8880) }";
> ```
> 
> There are currently two tests with the same name, you can remove them both and use this
> 
> ```rust
>     #[test]
>     fn private_key_debug_is_obfuscated() {
>         let sk =
>             "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy".parse::<PrivateKey>().unwrap();
>         let want =
>             "PrivateKey { compressed: true, network: Test, inner: SecretKey(#7217ac58fbad8880) }";
>         let got = format!("{:?}", sk);
>         assert_eq!(got, want)
>     }
> ```
> 
> The reason is we patched `impl_display_secret` macro during last scep release. Can you add a comment saying as such to the commit log please.

done

**Comment by tcharding** [comment]
Squash your commits bro, and remove formatting - otherwise I can't review. Thanks

**Comment by mahmudsudo** [comment]
> Squash your commits bro, and remove formatting - otherwise I can't review. Thanks

i am having issues squashing the commits because of the conflict merges , squashing is taking other peoples commit

**Comment by tcharding** [comment]
Well you are doing something else wrong. This PR should be your commits only. Perhaps you need to fetch `master` then rebase on that (and check you are fetching `upstream/master`).

**Comment by mahmudsudo** [comment]
> Well you are doing something else wrong. This PR should be your commits only. Perhaps you need to fetch `master` then rebase on that (and check you are fetching `upstream/master`).

will try that

**Comment by yancyribbens** [comment]
you need to rebase this branch on top of master.  Also, I recommend not naming your feature branch `master` since that further confuses the issue.

**Comment by mahmudsudo** [comment]
> you need to rebase this branch on top of master. Also, I recommend not naming your feature branch `master` since that further confuses the issue.

how about i close this pr and open another with clear commit and noting all the corrections

**Comment by yancyribbens** [comment]
> how about i close this pr and open another with clear commit and noting all the corrections

you could.  but you'd learn more if you figured out how to rebase your branch onto master.

**Pull request closed without merging**